### PR TITLE
Web, MacOS and Windows support

### PR DESCRIPTION
**Pull request by dirkbo**
Supporting macOS icons, mostly analog to iOS.

### CODE REVIEW

**Comment by watzon** [comment]
Any reason this hasn't been merged?

**Comment by dirkbo** [comment]
@watzon I think it was a bad moment, because shortly after my pull request they added support for flavors.

Today I updated from current flutter_launcher_icons:master with support for flavors and added flavors to macos as well.

**Comment by vishna** [comment]
@dirkbo great job, the only problem I noticed with this PR is it also adds extra files in the iOS folder (which I believe isn't intended)

**Comment by dirkbo** [comment]
@vishna That's odd, can you please tell me which files it touches in the ios Folder? (I won't have access to my mac until end of the week)

**Comment by Ephenodrom** [comment]
Any update on this?

**Comment by dirkbo** [comment]
@Ephenodrom I updated the mac version last week, should be OK to not touch iOS things anymore, and I added support for Windows.

**Comment by Ephenodrom** [comment]
Any reason this ist still not merged ?

**Comment by dirkbo** [comment]
There are no news on the pull request, I try to keep up with the master branch and I have an eye on the Web branch. 

As flutter Web now is stable, the focus should be on having web support first.

**Comment by felixwortmann** [comment]
This PR is open for >6 months. Flutter Web is now stable and Desktop is stable under early flag. What has to happen for this to be merged and published?

**Comment by Ephenodrom** [comment]
Can someone merge this and release a new version on pub.dev ?

**Comment by Ephenodrom** [comment]
@dirkbo Have you tested it? I tried out your repository and it seems to generate files like "app_icon_32.png.png". It seems not to trim the file type.

After I manually assigned the files within XCode it works, nur the Flutter icon is still shown in the launcher bar.

**Comment by dirkbo** [comment]
@Ephenodrom the double in the macos assets .png was a regression from merging with the @personalizedrefrigerator 's version to finally have web and desktop format together. I don't have access to  Mac all the time, so testing on Mac is an issue. I just pushed a new version, at least from the output files, now it looks ok.

**Comment by shaxxx** [comment]
Shout out to @MarkOSullivan94 - please merge this.

**Comment by shaxxx** [comment]
@dirkbo Just tested on MacOS Big Sur 11.4 and launcher icon is still original Flutter icon
My configuration (working on Android and IOS)
```
flutter_icons:
  image_path_android: "assets/icons/kasa_1024_1024_Android.png"
  image_path_ios: "assets/icons/kasa_1024_1024_iOS.png"
  image_path: "assets/icons/kasa_1024_1024_Android.png"
  macos: true
  windows: true
  android: true # can specify file name here e.g. "ic_launcher"
  ios: true # can specify file name here e.g. "My-Launcher-Icon"
  adaptive_icon_background: "#00CCFF" # only available for Android 8.0 devices and above
  adaptive_icon_foreground: "assets/icons/kasa_432_432_adaptive.png" # only available for Android 8.0 devices and above
```

Output I get 

>  ════════════════════════════════════════════
>      FLUTTER LAUNCHER ICONS (v0.9.1)                               
>   ════════════════════════════════════════════
>   
> • Creating adaptive icons Android
> • Updating colors.xml with color for adaptive icon background
> • Creating default icons Android
> • Overwriting the default Android launcher icon with a new icon
> • Overwriting default iOS launcher icon with new icon
> • Overwriting default windows launcher icon with new icon
> • Overwriting default macOS launcher icon with new icon
> 
> ✓ Successfully generated launcher icons

**Comment by dirkbo** [comment]
@shaxxx  Sorry for the late response, I'm currently embedded in a Project where I don't have access to a mac and not much time to work on this. I think I found the problem, when I merged web support from @personalizedrefrigerator's nicely refactored version, i accidentally chenged the macos default iconname. I can't test on mac before october, so please let me know if it works now.

**Comment by shaxxx** [comment]
@dirkbo I can confirm new version is working, well done!

> Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-x64, locale en-HR)

**Comment by sin2ned** [comment]
Any updates on this? MacOS and windows support would be so brilliant. Please merge and publish on pub.dev

**Comment by walsha2** [comment]
@dirkbo @MarkOSullivan94 any delay in getting this merged and pushed to pub.dev? Anything I can do help? I would prefer to not have to fork and just get this resolved in the main repo.

**Comment by westito** [comment]
This project seems dead. Somebody upload to pub.dev as "flutter_launcher_icons_v2"

**Comment by liugangnhm** [comment]
I think the PR is too big to merge

File: README.md [hunk 1]
```diff
@@ -39,6 +70,8 @@ dev_dependencies:
 flutter_icons:
   android: "launcher_icon"
   ios: true
+  windows: true
+  macos: false
```
Comment (krin-san) [inline README.md:74]: Let's also add `web: true` to indicate Web platform support.

File: example/default_example/pubspec.yaml [hunk 2]
```diff
@@ -19,6 +19,8 @@ flutter_icons:
   image_path_ios: "assets/images/icon-710x599-ios.png"
   android: true # can specify file name here e.g. "ic_launcher"
   ios: true # can specify file name here e.g. "My-Launcher-Icon"
+  macos: false # can specify file name here e.g. "My-Launcher-Icon"
+  windows: false # can specify file name here e.g. "My-Launcher-Icon"
```
Comment (krin-san) [inline example/default_example/pubspec.yaml:23]: Let's add `web: false` here too

File: example/flavors/ios/Flutter/Generated.xcconfig [hunk 3]
```diff
@@ -1,10 +1,14 @@
 // This is a generated file; do not edit or check into version control.
```
Comment (krin-san) [inline example/flavors/ios/Flutter/Generated.xcconfig:1]: Let's follow comment's suggestion and remove it from version control.

File: example/flavors/ios/Flutter/flutter_export_environment.sh [hunk 4]
```diff
@@ -1,11 +1,15 @@
 #!/bin/sh
 # This is a generated file; do not edit or check into version control.
```
Comment (krin-san) [inline example/flavors/ios/Flutter/flutter_export_environment.sh:2]: Let's follow comment's suggestion and remove it from version control.

File: lib/web.dart [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import 'package:flutter_launcher_icons/utils.dart';
+import 'package:image/image.dart';
+import 'package:flutter_launcher_icons/constants.dart' as constants;
+
+final IconTemplateGenerator iconGenerator =
+    IconTemplateGenerator(defaultLocation: constants.webIconLocation);
+
+List<IconTemplate> webIcons = <IconTemplate>[
+  iconGenerator.get(name: 'Icon-192.png', size: 192),
+  iconGenerator.get(name: 'Icon-512.png', size: 512),
```
Comment (krin-san) [inline lib/web.dart:13]: Latest web template also contains `Icon-maskable-192.png` and `Icon-maskable-512.png` assets. Can we adjust this file to generate these two as well?

File: README.md [hunk 6]
```diff
@@ -71,20 +104,28 @@ with the icon located in the image path specified above and given the name "laun
...
 
 - `image_path`: The location of the icon image file which you want to use as the app launcher icon
 
-- `image_path_android`: The location of the icon image file specific for Android platform (optional - if not defined then the image_path is used)
+- `image_path_android`: The location of the icon image file specific for Android platform (optional - if not defined then the `image_path` is used).
+
+- `image_path_ios`: The location of the icon image file specific for iOS platform (optional - if not defined then the `image_path` is used).
 
-- `image_path_ios`: The location of the icon image file specific for iOS platform (optional - if not defined then the image_path is used)
+- `image_path_web`: The location of the image file to be used for the web favicon and launcher (optional - if not defined, `image_path` is used).
```
Comment (cedvdb) [inline README.md:120]: This should be added to the readme.

**Comment by stem-lg** [comment]
what about linux ?

**Comment by Narven** [comment]
please use this:

https://pub.dev/packages/flutter_launcher_icons_maker

it works.

**Comment by felipecastrosales** [comment]
It's amazing how such a used and useful package still hasn't solved this problem. It's just a merge.

**Comment by RatakondalaArun** [comment]
Hi @dirkbo, 

Thanks for this great PR but unfortunately we had may changes since this PR and post that we added support for remaining platforms so we are closing this PR.

Once again thanks for your contribution ❤️

**Pull request closed without merging**